src/Target.cpp

```diff
@@ -753,6 +772,78 @@ int Target::natural_vector_size(const Halide::Type &t) const {
     }
 }
 
+bool Target::get_runtime_compatible_target(const Target& other, Target &result) {
+    const std::array<Feature, 25> union_features = {
+            Debug, TSAN, ASAN, MSAN, CUDA, CUDACapability30, CUDACapability32, CUDACapability35, CUDACapability50,
+            CUDACapability61, OpenCL, OpenGL, OpenGLCompute, MinGW, Metal, HexagonDma, HVX_64, HVX_128, HVX_v62,
+            HVX_v65, HVX_v66, HVX_shared_object, D3D12Compute, NoNEON, SoftFloatABI
+    };
+
+    const std::array<Feature, 12> intersection_features = {
+            SSE41, AVX, AVX2, FMA, FMA4, F16C, ARMv7s,VSX, AVX512, AVX512_KNL, AVX512_Skylake, AVX512_Cannonlake
+    };
+
+    // bitsets need to be the same width.
+    decltype(result.features) union_mask;
+    decltype(result.features) intersection_mask;
+
+    for (auto& feature : union_features) {
+        union_mask.set(feature);
+    }
+
+    for (auto& feature : intersection_features) {
+        intersection_mask.set(feature);
+    }
+
+    if (arch != other.arch || bits != other.bits || os != other.os) {
+        user_warning << "runtime targets must agree on platform (arch-bits-os)\n";
+        return false;
+    }
+
+    if (has_feature(SoftFloatABI) != other.has_feature(SoftFloatABI)) {
```

Comment:
Is this really the only feature that must match? IMHO some of the features in the 'union' set make more sense here (e.g., specifying `debug` for some-but-not-all of the targets should be an error, not a quietly-add-debug-to-everything).

Reply:
ok, so we should add a must_match std::array, and put SoftFloatABI and the debugging flags (debug, msan, tsan, asan) in it.